openadapt/app/tray.py

```diff
@@ -168,6 +183,67 @@ def _quit() -> None:
         # for storing toasts that should be manually removed
         self.sticky_toasts = {}
 
+    def stop_download(self) -> None:
+        global CANCEL_APP_DOWNLOAD
```

Comment:
What do you think about using a property instead of globals?

Reply:
@abrichr  We cannot use properties here.
Reason for this is that i am using `global` to communicate between multiple threads which cannot be done using properties as the **main thread** would be different and the download will happen in a **background thread**.

Even if i set the property in a **background thread** it wont be communicated to the **main thread**. So global would be the way to communicate between them

Reply:
I think we can use a `threading.Event` here, which is more thread-safe. e.g.:
```
class SystemTrayIcon(QObject):
    def __init__(self) -> None:
        self.cancel_download_event = threading.Event()

    def stop_download(self) -> None:
        self.cancel_download_event.set()

    @Slot(str)
    def download_start_toast_slot(self, message: str) -> None:
        self.cancel_download_event.clear()

    def download_latest_version(self, base_url: str, latest_version: str) -> None:
        DOWNLOAD_URL = ""
        FILE_NAME = ""

        if sys.platform == "darwin":
            FILE_NAME = f"OpenAdapt-v{latest_version}.app.zip"
            DOWNLOAD_URL = base_url + f"/v{latest_version}/{FILE_NAME}"
        else:
            FILE_NAME = f"OpenAdapt-v{latest_version}.zip"
            DOWNLOAD_URL = base_url + f"/v{latest_version}/{FILE_NAME}"

        downloads_path = os.path.join(os.path.expanduser("~"), "Downloads")
        local_filename = os.path.join(downloads_path, FILE_NAME)

        response = requests.get(DOWNLOAD_URL, stream=True)
        total_size = response.headers.get("content-length")
        total_size = int(total_size) if total_size else None
        block_size = 1024  # 1 Kilobyte
        start_time = time.time()
        last_update_time = start_time
        with open(local_filename, 'wb') as f:
            for data in response.iter_content(block_size):
                if self.cancel_download_event.is_set():
                    break
                f.write(data)
```

Reply:
Yess @abrichr this could be done in place of `global`
I have made the necessary change and it now uses `threading.Event()`.